psycopg3/psycopg3/waiting.py

```diff
@@ -71,23 +71,24 @@ def wait_conn(gen: PQGenConn[RV], timeout: Optional[float] = None) -> RV:
     :param gen: a generator performing database operations and yielding
         (fd, `Ready`) pairs when it would block.
     :param timeout: timeout (in seconds) to check for other interrupt, e.g.
-        to allow Ctrl-C.
+        to allow Ctrl-C. If zero or None, wait indefinitely.
     :type timeout: float
     :return: whatever *gen* returns on completion.
 
     Behave like in `wait()`, but take the fileno to wait from the generator
     itself, which might change during processing.
     """
+    timeout = timeout or None
     try:
         fileno, s = next(gen)
         sel = DefaultSelector()
         while 1:
             sel.register(fileno, s)
-            ready = None
-            while not ready:
-                ready = sel.select(timeout=timeout)
+            ready = sel.select(timeout=timeout)
             sel.unregister(fileno)
-            fileno, s = gen.send(ready[0][1])
+            if not ready:
+                raise e.OperationalError("timeout expired")
+            fileno, s = gen.send(ready[0][1])  # type: ignore[arg-type]
```

Comment:
Curious that this ignore is required. Or better, curious it wasn't required already. It seems a mypy bug: adding `reveal_type(ready)` after the innermost while on the master branch doesn't report anything.

Reply:
Agreed, the ignore should have been required already before.
We can get rid of it by changing to `fileno, s = gen.send(Ready(ready[0][1]))`, is it worth?

Reply:
Probably worth. I'm adding a small changeset just because I don't like these var names but I will leave the ignore around.